src/Forms/FieldList.php

```diff
@@ -389,9 +361,9 @@ public function hasTabSet()
      * Returns the specified tab object, if it exists
      *
      * @param string $tabName The tab to return, in the form "Tab.Subtab.Subsubtab".
-     * @return Tab|null The found or null
+     * @return Tab|Tabset|null The found Tab or Tabset, or null if nothing was found.
      */
-    public function findTab($tabName)
+    public function findTab(string $tabName): Tab|Tabset|null
```

Comment:
I think this and `findOrMakeTab` would return any `FormField` that happens to have that name - if that's the case, we should have an explicit exception thrown when that happens. Something like
```php
if ($currentPointer && !is_a($currentPointer, Tab::class) && !is_a($currentPointer, TabSet::class)) {
    $className = get_class($currentPointer);
    throw new RuntimeException("$tabName is an instance of '$className', not a tab");
}
```

Reply:
Also please check if it should be `Tabset` or `TabSet` - I think it's meant to be `TabSet`

Reply:
It's TabSet

We don't need to throw an exception as we now have strong return types will handle the wrong return type for us

Reply:
I'm on the fence about that - the error that PHP throws won't be very descriptive and might look like an application error in framework rather than the developer having put in the wrong name. I think throwing an exception here is worthwhile to make debugging that easier.

Reply:
Done